pinot-core/src/main/java/com/linkedin/pinot/core/startree/hll/HllConfig.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.startree.hll;
+
+import com.google.common.base.Preconditions;
+
+import java.util.*;
+
+public class HllConfig {
+
+    private int hllLog2m;
+    private Map<String, String> hllDeriveFieldNameMap;
+
+    public HllConfig() {
+        this(new ArrayList<String>());
+    }
+
+    public HllConfig(List<String> columnsToDeriveHllFields) {
+        this(columnsToDeriveHllFields, 5);
+    }
+
+    public HllConfig(List<String> columnsToDeriveHllFields, int hllLog2m) {
+        this(columnsToDeriveHllFields, hllLog2m, HllUtil.Constants.DEFAULT_HLL_DERIVE_COLUMN_SUFFIX);
+    }
+
+    public HllConfig(List<String> columnsToDeriveHllFields, int hllLog2m, String hllDeriveColumnSuffix) {
+        Preconditions.checkNotNull(columnsToDeriveHllFields, "ColumnsToDeriveHllFields should not be null.");
+        Preconditions.checkNotNull(hllDeriveColumnSuffix, "HLL Derived Field Suffix should not be null.");
```

Comment:
Good job with Preconditions!

Use Nonnull annotation on the paraemters for this and the constructor above